### PR TITLE
Replace 'userxx' with 'orgxx' in all test files when the user type is org 

### PR DESCRIPTION
**Pull request by nanguanlin6**
Currently 'userxx' and 'orgxx' are both used as username in test files when the user type is org, which is confusing. This PR replaces all 'userxx' with 'orgxx' when the user type is org(`user.type==1`). 
Some non-trivial changes
1. Rename `user3` dir to `org3` in `tests/git-repositories-meta` 
2. Change `end` in `issue reference` because 'org3' is one char shorter than 'user3'
ksnip_20230913-112819
3. Change the search result number of `user/repo2` because `user3/repo21` can't be searched now
ksnip_20230913-112931
4. Change the first org name getting from API because the result is ordered by alphabet asc and now `org 17` is before `org25`
JW8U7NIO(J$H _YCRB36H)T
)KFD411O4I8RB5ZOH7E0 Z3

Other modifications are just find all and replace all.
Unit tests with SQLite are all passed.

### CODE REVIEW

**Review by CaiCandong — APPROVED** [review]

**Review by CaiCandong — APPROVED** [review]

File: models/fixtures/user.yml
Comment (puni9869) [inline models/fixtures/user.yml]: Shoudn't we change this too for consistency.
```suggestion
  full_name: Org Six
```
I like these changes, we need more unit test cases in Gitea. Appreciated for your initiate.
I will take a closer look today. Could you please wait for my review. If it is on priority you can go ahead and merge after other maintainers approvals.

File: models/issues/issue_search.go
Comment (puni9869) [inline models/issues/issue_search.go]: Heads up, when doing changes in queries.

File: models/fixtures/user.yml
Comment (puni9869) [reply models/fixtures/user.yml]: Furture related :-  Could you slice this PR into multiple Pr. easy for me to keep track of changes, Thare are 90+ files. So.

**Comment by puni9869** [comment]
Does it required backport?

File: models/issues/issue_search.go
Comment (nanguanlin6) [reply models/issues/issue_search.go]: Thanks for your review. Those are from my other PR and I accidentally mixed them up.

Comment (puni9869) [reply models/issues/issue_search.go]: Could you point all of your PRs I can go one by one.

File: models/fixtures/user.yml
Comment (nanguanlin6) [reply models/fixtures/user.yml]: Thanks for your patient review. As I mentioned in the PR information, there are only three non-trivial changes that are not 'find and replace' in IDE. I don't think this can break into different parts because it's weird that both 'user3' and 'org3' are referred to the same user.

File: models/issues/issue_search.go
Comment (nanguanlin6) [reply models/issues/issue_search.go]: Never mind. I already deleted this.

File: models/fixtures/user.yml
Comment (nanguanlin6) [reply models/fixtures/user.yml]: > Shoudn't we change this too for consistency.
> 
> I like these changes, we need more unit test cases in Gitea. Appreciated for your initiate. I will take a closer look today. Could you please wait for my review. If it is on priority you can go ahead and merge after other maintainers approvals.

Not in a rush. Take your time :+1:

Comment (puni9869) [reply models/fixtures/user.yml]: >  I don't think this can break into different parts because it's weird that both 'user3' and 'org3' are referred to the same user.

You can use git blame to root this.

**Review by wxiaoguang — APPROVED** [review]
LGTM if tests pass

**Comment by CaiCandong** [comment]
[`8f20308` (#27052)](https://github.com/go-gitea/gitea/pull/27052/commits/8f2030892ccdf11fb9db026d5a135bdc3045fb76) @lng2020 Invited me to trigger a github actions check for him.

**Comment by lng2020** [comment]
The fix (https://github.com/go-gitea/gitea/pull/27052/commits/cdbfb2e819f779463d8f2a890301133f027e51a9) is because when getting all org from API, the result is ordered by alpha asc, and the test compares the first org full name. The older name `user17` is behind `org25`, and the newer name `org 17` is before `org25`. So the test should change.

**Review by silverwind — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Pull request merged**
